File: agrigrow-backend/routes/leaderboard.js
// routes/leaderboard.js — Top users ranked by points
const express = require('express');
const router = express.Router();
const auth = require('../middleware/auth');
const { dbAll } = require('../db/database');

// GET /api/leaderboard
router.get('/', auth, async (req, res) => {
  try {
    const top = await dbAll(
      "SELECT id, name, region, points FROM users WHERE role != 'admin' ORDER BY points DESC LIMIT 20",
      []
    );
    res.json({ leaderboard: top });
  } catch (err) {
    console.error('Leaderboard error:', err);
    res.status(500).json({ error: 'Failed to load leaderboard.' });
  }
});

module.exports = router;

File: flask-backend/routes/leaderboard.py
# routes/leaderboard.py — Top Users by Points
# GET /api/leaderboard  → Top 20 users ranked by points (admins excluded)

from flask import Blueprint, jsonify
from db.database import db_all
from middleware.auth_middleware import require_auth

leaderboard_bp = Blueprint('leaderboard', __name__)


# GET /api/leaderboard
@leaderboard_bp.route('/', methods=['GET'])
@require_auth
def get_leaderboard():
    try:
        top = db_all(
            "SELECT id, name, region, points FROM users WHERE role != 'admin' ORDER BY points DESC LIMIT 20"
        )
        return jsonify({'leaderboard': top})

    except Exception as e:
        print('Leaderboard error:', e)
        return jsonify({'error': 'Failed to load leaderboard.'}), 500
